lib/iris/cube.py

```diff
@@ -3131,6 +3084,10 @@ def __eq__(self, other):
         if isinstance(other, Cube):
             result = self.metadata == other.metadata
 
+            # check the cube fill-value.
+            if result:
+                result = self.fill_value == other.fill_value
```

Comment:
> previously we didn't account for `dtype` when performing equality

Or indeed for the data array... and the `dtype` is part of the data array, so maybe you're right.

Alternatively, two cubes will not merge if they have different dtypes, so perhaps we _should_ consider `dtype` when comparing for equality??